configure

```diff
@@ -0,0 +1,360 @@
+        if test -n "$compiler"; then
+            sub_info "$compiler"
+            break # Found
+        fi
+    done
+    info Checking Host C++ Compiler...
+    if test -n "$BUILD"; then
+        for guess in "$BUILD"-g++ "$BUILD"-clang++ "$BUILD"-c++; do
+            findcxx "$guess" host
+            if test -n "$hostcompiler"; then
+                sub_info "$hostcompiler"
+                break # Found
+            fi
+        done
+    else
+        for guess in g++ clang++ c++; do
+        findcxx "$guess" host
+        if test -n "$hostcompiler"; then
+            sub_info "$hostcompiler"
+            break # Found
+        fi
+        done
+    fi
+    if test -z "$compiler" || test -z "$hostcompiler"; then
+       sub_error No C++ compiler found! # Not found
+    fi
+else
+    for guess in g++ clang++ c++; do
+        findcxx "$guess"
+        if test -n "$compiler"; then
+            sub_info "$compiler"
+            break # Found
+        fi
+    done
+    if test -z "$compiler"; then
+       sub_error No C++ compiler found! # Not found
+    fi
+fi
+cxx_works
+
+## Verify PLATFORM value
+if [ "$PLATFORM" != "Linux" ] && [ "$PLATFORM" != "FreeBSD" ] && \
+[ "$PLATFORM" != "OpenBSD" ] && [ "$PLATFORM" != "Darwin" ]; then
+    warning "$PLATFORM" platform is unknown!
+    sub_warning Valid Platforms are: Linux, FreeBSD, OpenBSD, Darwin
+fi
+
+## Find depends
+for depend in gmake m4 install; do
+    info Checking "$depend" is executable...
+    find_program $depend
+done
+
+## Test compiler/linker supported arguments
+if test -z $NO_CXXOPTS; then
+    for argument in -std=c++11 \
+                    -Wall \
+                    -Os \
+                    -fno-plt
+    do
+        try_cxx_argument CXXOPTS $argument
+    done
+    if [ "$PLATFORM" != "FreeBSD" ]; then
+        try_cxx_argument CXXOPTS -flto
+    fi
+    if [ "$PLATFORM" != "Darwin" ]; then
+        try_cxx_argument CXXOPTS -fno-rtti
+    fi
+fi
+if [ "$PLATFORM" = "FreeBSD" ] && test -z $NO_LDFLAGS; then
+    try_ld_argument LDFLAGS -lrt
+fi
+if [ "$NO_SANITIZERS" = "auto" ]; then
+    try_ld_argument SANITIZEROPTS -fsanitize=address,undefined
+elif [ "$NO_SANITIZERS" = "0" ]; then
+    SANITIZEROPTS="-fsanitize=address,undefined"
+fi
+
+## Create mconfig
+rm -f testfile*
+info Creating mconfig...
+cat << EOF > mconfig
+## Auto-generated by "$0" for "$PLATFORM"
+# Installation path options.
+
+SBINDIR=$SBINDIR
+MANDIR=$MANDIR
+SYSCONTROLSOCKET=$SYSCONTROLSOCKET
+
+# General build options.
+
+# Linux (GCC): Note with GCC 5.x/6.x you must use the old ABI, with GCC 7.x you must use
+# the new ABI. See BUILD file for more information.
+# MacOS: Cannot use -fno-rtti: apparently prevents exception handling from working properly.
+# FreeBSD: Cannot use LTO with default linker.
+
+CXX=$compiler
+CPPFLAGS=$CPPFLAGS
+CXXOPTS=$CXXOPTS
+CXXFLAGS=$CXXFLAGS
+LDFLAGS=$LDFLAGS
+CUSTOM_LDFLAGS=$CUSTOM_LDFLAGS
+BUILD_SHUTDOWN=$BUILD_SHUTDOWN
+SANITIZEOPTS=$SANITIZEROPTS
+
+# Notes:
```

Comment:
you probably shouldn't generate these comments as they may not be applicable anyway.